app.js

```diff
@@ -12,7 +13,41 @@ app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+  });
+  
+  // Retrieve an access token
+  spotifyApi
+    .clientCredentialsGrant()
+    .then(data => spotifyApi.setAccessToken(data.body['access_token']))
+    .catch(error => console.log('Something went wrong when retrieving an access token', error));
 
 // Our routes go here:
+app.get('/', (req, res)=>{
+    res.render('home');
+})
+
+app.get('/artist-search',  async (req, res)=>{
+    try{ 
+        let data = await spotifyApi.searchArtists(req.query.artist);
+        console.log('The received data from the API: ', data.body.artists.items);
+        res.render('artist', {results:data.body.artists.items});
+    } 
+    catch (error) {
+        console.log('The error while searching artists occurred: ', error)
+    } 
+});
+
+app.get('/albums/:artistId', async (req, res, next) => {
+    try{
+       let data = await spotifyApi.getArtistAlbums(req.params.artist)
```

Comment:
Since you wrote that the `URL` to this `route` is `'/albums/:artistId'`, you need to link the params with your route for this call to the `API`  to work properly. 
```let data = await spotifyApi.getArtistAlbums(req.params.artistId)```